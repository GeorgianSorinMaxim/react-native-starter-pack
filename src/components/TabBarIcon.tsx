import * as React from 'react';
import Icon from 'react-native-vector-icons/FontAwesome';

import Colors from '../constants/Colors';

type IconProps = {
  name: string,
  focused: boolean
};

export default function TabBarIcon(props: IconProps) {
  return (
    <Icon
      size={20}
      name={props.name}
      color={props.focused ? Colors.tabIconSelected : Colors.tabIconDefault}
    />
  );
}
