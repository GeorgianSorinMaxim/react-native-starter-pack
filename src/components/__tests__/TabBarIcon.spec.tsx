import * as React from "react";
import renderer from "react-test-renderer";

import TabBarIcon from "../TabBarIcon";

it(`renders correctly`, () => {
  const defaultProps = {
    name: "map",
    focused: true
  };
  const tree = renderer.create(<TabBarIcon {...defaultProps} />).toJSON();

  expect(tree).toMatchSnapshot();
});
