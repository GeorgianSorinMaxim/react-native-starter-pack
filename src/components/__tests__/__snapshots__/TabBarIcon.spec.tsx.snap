// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#2f95dc",
        "fontSize": 20,
      },
      undefined,
      Object {
        "fontFamily": "FontAwesome",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  ï‰¹
</Text>
`;
